psdisk/psdisk.c

```diff
@@ -171,50 +585,130 @@ static int psdisk_createImg(int optind, char *argv[])
 
 static void psdisk_destroy(void)
 {
+	struct plist_node_t *node;
+
+	while (!LIST_EMPTY(&psdisk_common.list)) {
+			node = LIST_FIRST(&psdisk_common.list);
+			LIST_REMOVE(node, ptrs);
+			free(node);
+	}
+
 	if (psdisk_common.file != NULL)
 		fclose(psdisk_common.file);
 }
 
 
+static int psdisk_handleImg(void)
+{
+	int res = -1;
+	int rmFile = 0;
+
+	if (CREATE_IMG(psdisk_common.opts)) {
+		if ((res = psdisk_createImg()) < 0) {
+			printf("Cannot create file system image '%s'.\n", psdisk_common.fileName);
+			rmFile = 1;
+		}
+		else {
+			printf("File system image '%s' was created correctly.\n", psdisk_common.fileName);
+		}
+	}
+	else if (UPDATE_IMG(psdisk_common.opts)) {
+		if ((res = psdisk_updateImg()) < 0)
+			printf("Cannot update file system image '%s'.\n", psdisk_common.fileName);
+		else
+			printf("File system image '%s' was updated correctly.\n", psdisk_common.fileName);
+
+	}
+	else if (READ_IMG(psdisk_common.opts)) {
+		if ((res = psdisk_readPartsTable()) < 0)
+			printf("Cannot read file system image '%s'.\n", psdisk_common.fileName);
+		else
+			psdisk_showPartsTable();
+	}
+	else {
+		printf("Inappropriate option, read help.\n");
+		rmFile = !FILE_EXIST(psdisk_common.opts);
+	}
+
+	psdisk_destroy();
+	if (rmFile)
+		remove(psdisk_common.fileName);
+
+	return res;
+}
+
+
 int main(int argc, char *argv[])
 {
 	int opt;
 
-	while ((opt = getopt(argc, argv, "i:s:p:h")) != -1) {
+	psdisk_common.pCnt = 0;
+	psdisk_common.opts = 0;
+	LIST_INIT(&psdisk_common.list);
+
+	if (argc < 2) {
+		printf("%s: bad usage\n", argv[0]);
+		printf("Try '%s -h' for more information.\n", argv[0]);
+		return -1;
```

Comment:
Just print help

Reply:
The same case as in `psdisk_handleImg`. To be consisted with the other command line tools the information about incorrect usage is printed.